WIKI.org

```diff
@@ -52,79 +52,80 @@ to your Dropbox, here are your options:
-
- [[https://github.com/200ok-ch/organice/wiki/videos/demo-webdav-failing-cors.gif]]
-
-*** Setup your own WebDAV Server with Apache2 on Debian
-
- For testing purposes, here are the instructions to setup WebDAV
- locally on your machine using Apache2 using Debian. These instructions
- are not meant to be used in production, though.
-
- *Initial package installation*
-
- #+BEGIN_SRC shell
- sudo apt -y install apache2-utils apache2
- #+END_SRC
-
- *Set up a new vhost for webdav*
-
- =/etc/apache2/sites-available/webdav.conf=
-
- #+BEGIN_EXAMPLE
- Alias /webdav /srv/dav/
-
- RewriteEngine On
- RewriteCond %{REQUEST_METHOD} OPTIONS
- RewriteRule ^(.*)$ $1 [R=200,L]
-
- <Location /webdav>
-     Options Indexes
-     DAV On
-     AuthType Basic
-     AuthName "webdav"
-     AuthUserFile /srv/dav/.htpasswd
-     Require valid-user
-
-     Header always set Access-Control-Allow-Origin "*"
-     Header always set Access-Control-Allow-Methods "GET,POST,OPTIONS,DELETE,PUT,PROPFIND"
-     Header always set Access-Control-Allow-Headers "Origin,Accept,X-Requested-With,Content-Type,Access-Control-Request-Method,Access-Control-Request-Headers,Authorization,X-CSRF-Token,Depth"
-     Header always set Access-Control-Allow-Credentials true
-
-     Require all granted
- </location>
- #+END_EXAMPLE
-
- *Enable Apache modules*
-
- #+BEGIN_SRC shell
- sudo a2enmod headers
- sudo a2enmod dav*
- sudo a2enmod rewrite
- sudo a2ensite webdav
- #+END_SRC
-
- *Setup folder, password and rights*
-
- #+BEGIN_SRC shell
- sudo mkdir /srv/dav
- sudo htpasswd -c /srv/dav/.htpasswd webdav
- sudo chmod 770 /srv/dav; sudo chown www-data. /srv/dav
- sudo service apache2 restart
- #+END_SRC
-
- *Test webdav access using a commandline tool*
-
- #+BEGIN_SRC shell
- sudo apt -y install cadaver
- cadaver http://localhost/webdav/
- #+END_SRC
+*** WebDAV test server
+
+For testing purposes, we use a Docker image with a proven and well
+documented server: Apache2 running on Debian. You can build the Docker
+image yourself - and customize the Apache configuration to your needs:
+
+#+BEGIN_SRC shell
+docker build -f doc/webdav/Dockerfile -t apache-webdav .
+#+END_SRC
+
+Then run the Apache2 WebDAV server with:
+
+#+BEGIN_SRC shell
+docker run -dit --name apache-webdav-app -p 8080:80 apache-webdav
+#+END_SRC
+
+On your host machine, you can now login with any WebDAV client using
+the URL =http://localhost:8080/webdav=. There is no authentication
+configuration, so any user account works (including omitted user
+accounts). It goes without saying that if you wanted to use this for
+production, please enable authentication. Within the test image,
+you'll find the [[file:sample.org][sample.org]] file, so you can get started developing and
+testing right away.
+
+For testing WebDAV outside of organice, and you're an Emacs user, you
+can open [[/dav:localhost#8080:/webdav/][this]] link (=C-c C-o=). Then, you will get a [[https://www.gnu.org/software/tramp/][TRAMP]] session
```

Comment:
Very nice Org mode examples here :) I just discovered TRAMP and babel (shell) and now use it a lot in my org files, e.g. to generate hledger reports right in Org mode :)

Reply:
Thanks^^

Yes, TRAMP and Babel source blocks are mighty powerful - I'm using them all the time. At 200ok, we also use Org in combination with ledger(; We also use it for controlling, writing quotes and invoices. With such literate docs, we gain transparency and reproducability. Plus, it's just a lot more fun to do these chores like that(;

Reply:
Oooh... I knew about TRAMP and babel, but hledger is a very cool new find for me :laughing:  Thanks!

Reply:
If you're into text based accounting (and who wouldn't want to use version control on serious money related stuff?), maybe I can interest you in another FOSS project of ours: easy.

(h)ledger only solves the data format and versioning problem. However, the entries are still mutable state. We've written a system that's adding event orientation to the game. Doing bookkeeping like that has been a game changer for us. Another additional benefit is that I (as a business owner) don't really want to think in accounting terms, I want to think in 'business event' terms (writing invoices, paying invoices, paying employees and so forth). With easy, I can actually focus on that and it takes care of the rest. 

https://github.com/200ok-ch/easy